created and organized folders and files for base framework

bubsinthemountains opened the pull request:
created general react framework with folders and files; added test folder to hold future test files; split 4 components into separate folders within component folder

h-sheeva commented:
@bubsinthemountains - your PR mentions you created 4 separate folders?  Git won't push folders if they have empty content, so I cannot see any folders outside of the components folder which only contains an empty App file.  If you do have folders you'd like to push, you can add an empty ".gitinclude" file to each folder so that you can commit.

bubsinthemountains commented:
@h-sheeva i added .gitinclude to the folders and added a spec.test.js file to the test folder. can you confirm you see the four widget folders inside component? thanks.

h-sheeva reviewed (approved):
lgtm!
